backend/src/components/reports.js

```diff
@@ -41,6 +52,139 @@ async function getFacilityReports(req, res) {
   }
 }
 
+async function getSurveySections(req, res) {
+  try {
+    const sections = []
+    const operation = `ofm_sections?$filter=ofm_is_published eq true and _ofm_survey_value eq '${req?.query?.surveyId}'&$orderby=ofm_section_order`
+    const response = await getOperation(operation)
+    response?.value?.forEach((section) => sections.push(new MappableObjectForFront(section, SurveySectionMappings).toJSON()))
+    return res.status(HttpStatus.OK).json(sections)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function getSurveyQuestions(req, res) {
+  try {
+    const questions = []
+    let operation
+    if (req?.query?.sectionId) {
+      operation = `ofm_questions?$select=ofm_question_choice,ofm_question_id,ofm_question_text,ofm_question_type,ofm_sequence,ofm_fixed_response,_ofm_header_value,ofm_maximum_rows,ofm_response_required,ofm_occurence&$expand=ofm_question_business_rule_ques($select=_ofm_true_child_question_value,_ofm_false_child_question_value,ofm_question_business_ruleid,ofm_condition,ofm_parent_has_response,_ofm_child_question_value)&$filter=ofm_is_published eq true and _ofm_section_value eq '${req?.query?.sectionId}'&$orderby=ofm_sequence`
+    }
+    const response = await getOperation(operation)
+    response?.value?.forEach((question) => questions.push(mapQuestionObjectForFront(question)))
+    return res.status(HttpStatus.OK).json(questions)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function createSurveyResponse(req, res) {
+  try {
+    const payload = {
+      'ofm_contact@odata.bind': `/contacts(${req.body?.contactId})`,
+      'ofm_facility@odata.bind': `/accounts(${req.body?.facilityId})`,
+      'ofm_survey@odata.bind': `/ofm_surveies(${req.body?.surveyId})`,
```

Comment:
Is **ofm_surveies** a typo or some Dynamics weirdness?

Reply:
it is a Dynamics term for "surveys"

Reply:
Blame Dynamics then!